src/xvimage.c

```diff
@@ -1448,6 +1448,10 @@ void KillOldPics(void)
   if (picComments) free(picComments);
   picComments = (char *) NULL;
   ChangeCommentText();
+
+  if (picExifInfo) free(picExifInfo);
+  picExifInfo = (byte *) NULL;
+  picExifInfoSize = 0;
 }
```

Comment:
I think that the real issue was that KillOldPics() didn't free picExifInfo so if the last image that you opened happened to be an jpeg with exif info, it would be leaked.